underfs/local/src/main/java/alluxio/underfs/local/LocalUnderFileSystem.java

```diff
@@ -289,6 +291,21 @@ public void connectFromWorker(String hostname) throws IOException {
   }
 
   /**
+   * Rename a file to a file or a directory to a directory.
+   *
+   * @param src path of source file or directory
+   * @param dst path of destination file or directory
+   * @return true if rename succeeds
+   * @throws IOException
+   */
+  private boolean rename(String src, String dst) throws IOException {
```

Comment:
I favor keeping the semantics consistent between under storages, especially since the overhead should be low for these cases. Otherwise we may run into strange edge cases due to different behavior.